rust_snuba/src/processors/profile_chunks.rs

```diff
@@ -15,26 +15,19 @@ pub fn process_message(
     config: &ProcessorConfig,
 ) -> anyhow::Result<InsertBatch> {
     let payload_bytes = payload.payload().context("Expected payload")?;
-    let msg: InputMessage = serde_json::from_slice(payload_bytes)?;
+    let msg: FromChunkMessage = serde_json::from_slice(payload_bytes)?;
+    let origin_timestamp = DateTime::from_timestamp(msg.received, 0);
+    let mut chunk: Chunk = msg.try_into()?;
 
-    let mut row = Chunk {
-        chunk: msg,
-        offset: metadata.offset,
-        partition: metadata.partition,
-    };
+    chunk.offset = metadata.offset;
+    chunk.partition = metadata.partition;
+    chunk.retention_days = Some(enforce_retention(chunk.retention_days, &config.env_config));
```

Comment:
Might just be a personal preference, but having an infallible `TryInto` implementation, or one that only has a single caller is quite unnecessary.
You can just define the `Chunk` struct right here.